app/classifier/frame-annotator.jsx

```diff
@@ -0,0 +1,282 @@
+  eventCoordsToSVGCoords(x, y) {
+    const svg = this.refs.svgSubjectArea;
+    const newPoint = svg.createSVGPoint();
+    newPoint.x = x;
+    newPoint.y = y;
+    const matrixForWindowCoordsToSVGUserSpaceCoords = this.getMatrixForWindowCoordsToSVGUserSpaceCoords();
+    const pointforSVGSystem = newPoint.matrixTransform(matrixForWindowCoordsToSVGUserSpaceCoords);
+    return pointforSVGSystem;
+  }
+
+  handleAnnotationChange(oldAnnotation, currentAnnotation) {
+    if (oldAnnotation) {
+      const lastTask = this.props.workflow.tasks[oldAnnotation.task];
+      const LastTaskComponent = tasks[lastTask.type];
+      if (LastTaskComponent.onLeaveAnnotation) {
+        return LastTaskComponent.onLeaveAnnotation(lastTask, oldAnnotation);
+      }
+    }
+  }
+
+  render() {
+    let warningBanner;
+    let taskDescription;
+    let TaskComponent;
+    let BeforeSubject;
+    let InsideSubject;
+    let AfterSubject;
+    const { type, src } = getSubjectLocation(this.props.subject, this.props.frame);
+    const createdViewBox = `${this.props.viewBoxDimensions.x} ${this.props.viewBoxDimensions.y} ${this.props.viewBoxDimensions.width} ${this.props.viewBoxDimensions.height}`;
+
+    if (this.props.annotation) {
+      taskDescription = this.props.workflow.tasks[this.props.annotation.task];
+      TaskComponent = tasks[taskDescription.type];
+    }
+
+    if (this.state.alreadySeen) {
+      warningBanner = (
+        <WarningBanner label="Already seen!">
+          <p>Our records show that you’ve already seen this image. We might have run out of data for you in this workflow!</p>
+          <p>Try choosing a different workflow or contributing to a different project.</p>
+        </WarningBanner>
+      );
+    } else if (this.props.subject.retired) {
+      warningBanner = (
+        <WarningBanner label="Finished!">
+          <p>This subject already has enough classifications, so yours won’t be used in its analysis!</p>
+          <p>If you’re looking to help, try choosing a different workflow or contributing to a different project.</p>
+        </WarningBanner>
+      );
+    }
+
+    const svgStyle = {};
+    if (type === 'image' && !this.props.loading) {
+      // Images are rendered again within the SVG itself.
+      // When cropped right next to the edge of the image,
+      // the original tag can show through, so fill the SVG to cover it.
+      svgStyle.background = 'black';
+
+      // Allow touch scrolling on subject for mobile and tablets
+      if (taskDescription) {
+        if ((taskDescription.type !== 'drawing') && (taskDescription.type !== 'crop')) {
+          svgStyle.pointerEvents = 'none';
+        }
+      }
+      if (this.props.panEnabled === true) {
+        svgStyle.pointerEvents = 'all';
+      }
+    }
+
+    const svgProps = {};
+
+    if (TaskComponent) {
+      BeforeSubject = TaskComponent.BeforeSubject;
+      InsideSubject = TaskComponent.InsideSubject;
+      AfterSubject = TaskComponent.AfterSubject;
```

Comment:
Webpack complains about this syntax and won't rebuild. I've tried `const {BeforeSubject, InsideSubject, AfterSubject} = TaskComponent;`, but this fails because it needs to check that `TaskComponent` has been defined. It doesn't seem to like using destructuring assignment in a conditional.